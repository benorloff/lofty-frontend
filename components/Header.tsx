import { useState } from "react";
import { 
    Container, 
    AppBar, 
    Toolbar, 
    Typography,
    Button,
    IconButton,
    Avatar,
} from "@mui/material";
import UploadModal from "./UploadModal";
import Link from "next/link";
import { useSession, signIn, signOut } from "next-auth/react";

export default function Header() {

    const { data: session, status } = useSession()
    const [modalOpen, setModalOpen] = useState(false)
    
    console.log(session, '<-- session data')

    const handleModalOpen = () => {
        setModalOpen(true)
    }

    const handleModalClose = () => {
        setModalOpen(false)
    }

    return (
        <>
            <AppBar component="nav" position="sticky">
                <Container maxWidth="lg">
                    <Toolbar disableGutters>
                        <Typography 
                            variant="h5" 
                            component="div"
                            sx={{ flexGrow: 1, ml: 1 }}
                        >
                            <Link href={'/'}>Catstagram</Link>
                        </Typography>
                        {/* Get session status and display actions accordingly */}
                        { status === "authenticated" ? (
                            <>
                                <Button 
                                    variant="contained"
                                    onClick={() => handleModalOpen()}
                                    sx={{ mr: 1 }}
                                >
                                    Add Photo
                                </Button>
                                <IconButton
                                    sx={{ mr: 1 }}
                                >
                                    <Avatar alt={`${session?.user?.name}`} src={`${session?.user?.image}`} />
                                </IconButton>
                                <Button
                                    variant="contained"
                                    onClick={() => signOut()}
                                >
                                    Log Out
                                </Button>
                            </>
                        ) : (
                            <>
                                <Link href="/register">
                                    <Button variant="contained" sx={{ mr: 2 }}>
                                        Sign Up
                                    </Button>
                                </Link>
                                <Button
                                    variant="contained"
                                    onClick={() => signIn()}
                                >
                                    Log In
                                </Button>
                            </>
                            
                        )}
                    </Toolbar>
                </Container>
            </AppBar>
            <UploadModal open={modalOpen} handleClose={handleModalClose} />
        </>
    )
}