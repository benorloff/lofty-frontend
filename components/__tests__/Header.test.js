import Header from '../Header'
import { render, screen } from "@testing-library/react"
import "@testing-library/jest-dom"

jest.mock('next/router', () => require('next-router-mock'))

jest.mock("next-auth/react", () => {
    const originalModule = jest.requireActual('next-auth/react')
    const mockSession = {
        expires: new Date(Date.now() + 2 * 86400).toISOString(),
        user: { username: "admin"}
    }
    return {
        __esModule: true,
        ...originalModule,
        useSession: jest.fn(() => {
            return { data: mockSession, status: 'unauthenticated'}
        })
    }
})

describe('Header Component', () => {

    it('Show Log In when no session', async () => {

        const { container } = render(<Header />)

        expect(container).toMatchSnapshot()
        expect(screen.getByText("Log In")).toBeInTheDocument()

    })
})