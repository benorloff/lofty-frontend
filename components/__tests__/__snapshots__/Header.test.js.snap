// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Show Log In when no session 1`] = `
<div>
  <nav
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-7e0s21-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 css-1k7hxd6-MuiTypography-root"
        >
          <a
            href="/"
          >
            Catstagram
          </a>
        </div>
        <a
          href="/register"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1s5vo3j-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Sign Up
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </a>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
          name="Sign In"
          tabindex="0"
          type="button"
        >
          Log In
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </nav>
</div>
`;
